File: SDK/python/verifio/verify.py
from typing import Optional, cast, Dict, Any
from .client import VerifioClient
from .types import VerificationResult, VerifyOptions

class VerifyService:
    def __init__(self, client: VerifioClient):
        self.client = client

    def verify(self, email: str, options: Optional[VerifyOptions] = None) -> VerificationResult:
        body: Dict[str, Any] = {"email": email}
        if options:
            body["options"] = options
            
        response_data = self.client.request("POST", "/verify", body)
        
        if not response_data.get("success") or not response_data.get("data"):
            raise Exception(response_data.get("error") or "Verification failed")
            
        return response_data["data"]

File: SDK/rust/src/verify.rs
use reqwest::Method;
use serde::Serialize;
use std::sync::Arc;

use crate::client::VerifioClient;
use crate::error::VerifioError;
use crate::types::{VerificationResult, VerifyOptions};

#[derive(Clone, Debug)]
pub struct VerifyService {
    client: Arc<VerifioClient>,
}

impl VerifyService {
    pub(crate) fn new(client: Arc<VerifioClient>) -> Self {
        Self { client }
    }

    pub async fn verify(
        &self,
        email: &str,
        options: Option<VerifyOptions>,
    ) -> Result<VerificationResult, VerifioError> {
        #[derive(Serialize)]
        struct VerifyRequest<'a> {
            email: &'a str,
            #[serde(skip_serializing_if = "Option::is_none")]
            options: Option<VerifyOptions>,
        }

        let body = VerifyRequest { email, options };

        self.client
            .request(Method::POST, "/verify", Some(&body))
            .await
    }
}
